### ISSUE TITLE
`wails serve` not shutting down after waking computer up from sleep

### ISSUE THREAD

**Issue by pedzed** [wailsapp/wails#415]
**Description**
After running `wails serve` for some time, it is not possible to quit it (or it takes extremely long?) using `CTRL+C`.

**To Reproduce**
1. Run `wail serve`
1. ~Develop for a while~
1. Put your computer to sleep and then awake
1. Try to quit using `CTRL+C`

**Expected behaviour**
The program should quit within 5 seconds.

**System Details**

| Name   | Value |
| ----- | ----- |
| Wails Version | v1.5.0 |
| Go Version    | go1.14.2 |
| Platform      | darwin (macOS) |
| Arch          | amd64 |
| GO111MODULE   | on |
| GCC           | 4.2.1 |
| Npm           | 6.14.5 |
| Node          | v14.2.0 |

**Additional context**
These are the last lines of the terminal output:
```
DEBU[9942] [Events] Got Event                            data="[]" name="wails:loaded"
^CDEBU[11436] [App] SIGNAL CAUGHT! Starting Shutdown
DEBU[11436] [Bridge] Shutting down
```
From there, it just hangs.

**Comment by leaanthony** [comment]
Is your project online so we can look at it?

**Comment by pedzed** [comment]
> Is your project online so we can look at it?

Whether the code is open source? No, but I am planning on open sourcing it after it comes close to an alpha stage. I have invited you to the repo. If others would like to chime in, you're welcome to be my guest.

**Comment by pedzed** [comment]
For the sake of this issue, I tried reproducing it and my gut feeling was right: it is (100% of the times) reproducible if the computer is put asleep and then turned on again.

This time however, I see that there is a broken pipe in the end:
```
DEBU[0018] [Events] Got Event                            data="[]" name="wails:loaded"
DEBU[0018] [BridgeSession] write tcp [::1]:34115->[::1]:64557: write: broken pipe
^CDEBU[0047] [App] SIGNAL CAUGHT! Starting Shutdown
DEBU[0047] [Bridge] Shutting down
```

`wails serve`-ing again after forcefully exiting makes it normally exitable again. When the computer is put asleep and then gets turned on, it hangs again.

**Comment by tmclane** [comment]
Can you provide access to a simplified application that demonstrates the issue?

**Comment by pedzed** [comment]
@tmclane I have added you to the repo.

**Comment by tmclane** [comment]
It sounds like the goroutine for that session that was alive before the machine went to sleep is not exiting in this case.

**Comment by tmclane** [comment]
I'm sorry I haven't had time to look at this lately.  I'm planning on investigating and adding a fix soon.  In the meantime I would recommend exiting the `wails serve` before sleeping.

**Comment by tmclane** [comment]
Although your description itself talks about `Ctrl-C` the title is `waking from sleep`.

Are both an issue?

**Comment by pedzed** [comment]
> In the meantime I would recommend exiting the `wails serve` before sleeping.

Of course, but I often forget it.

> Although your description itself talks about `Ctrl-C` the title is `waking from sleep`.
> 
> Are both an issue?

I thought my issue was clear enough. See the heading 'To Reproduce'. If you have a better title in mind, you're welcome to edit it or provide a suggestion.

**Comment by tmclane** [comment]
Just looking for all the information I could get is all.  Thanks for the clarification!

**Comment by JokerQyou** [comment]
This issue still persists on macOS Big Sur. If `wails serve` is running and the computer was put into sleep and woke up, `Ctrl-C` won't be able to shutdown wails process. If you manually kill the process either with `kill -9` or via `Activity Monitor`, port 8080 won't be released, so another `wail serve` would fail. The only thing you can do is to reboot. This is very frustrating during development.

**Comment by leaanthony** [comment]
Thanks for reopening this. There's nothing inherently different about a Wails app and any other Go app so this must have been seen in other apps. What have you done in the way of debugging this?

**Comment by JokerQyou** [comment]
To be clear, I believe this is not an issue with a Wails app, but an issue with `wails serve` command. That is - we generally don't `Ctrl-C` or send `INT` signal to a deployed GUI app. But since `wails serve` is a command used in the terminal, during development stage, so it will be very common for the developer to interrupt it. I haven't got the time to debug this, actually.

**Comment by charlie-dee** [comment]
I have come across this issue today (macOS, M1). Quite bizarre, I have not had an issue over the last x months using wails, today I updated to Go 1.16, immediately blamed that, however the undesirable behaviour persisted.  

I  tried killing all processes associated with my app, however it has dawned upon me that Safari was the problem this evening.

Historically I've used Brave, but with all the goodies shown off at WWDC 👀 I decided to give Safari another shot, going back to Brave for development solved the problem. 

Debug shown when using Safari: `[BridgeSession] write tcp [::1]:34115->[::1]:49333: write: broken pipe `
